File: depend/libcmb/perl/tests/test2.pl
#!/usr/bin/env perl
use strict;
use warnings;
use Cmb;

our $total = 0;
my $choice = 2;
my $num = 10000;

my @items = (1..$num);
my $cmb = new Cmb { size_min => $choice, size_max => $choice };

printf "Silently enumerating choose-%u from %u:\n", $choice, $num;
$cmb->cmb_callback($#items + 1, \@items, sub { $total++; return 0 });
printf "%u callbacks executed\n", $total;

File: depend/libcmb/tests/test2.c
/*-
 * Copyright (c) 2018-2026 Devin Teske <dteske@FreeBSD.org>
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */

#include <cmb.h>
#include <stdio.h>
#include <stdlib.h>

#define CHOICE 2
#define NITEMS 10000

static int total = 0;

static int
afunc(struct cmb_config *config, uint64_t seq, uint32_t nitems, char *items[])
{
	total++;
	return (0);
}

int
main(void)
{
	static struct cmb_config config = {
		.size_min = CHOICE,
		.size_max = CHOICE,
		.action = afunc,
	};
	char *items[NITEMS];

	printf("Silently enumerating choose-%u from %u:\n", CHOICE, NITEMS);
	(void)cmb(&config, NITEMS, items);

	return (EXIT_SUCCESS);
}
